handsontable/src/shortcuts/context.js

```diff
@@ -0,0 +1,189 @@
+      variants,
+      callback,
+      namespace,
+      runAction = () => true,
+      preventDefault = true,
+      stopPropagation = false,
+      relativeToNamespace = '',
+      position = 'after',
+    } = {}) => {
+
+    if (isUndefined(namespace)) {
+      throw new Error('Please define a namespace for added shortcut.');
+    }
+
+    if (isFunction(callback) === false) {
+      throw new Error('Please define a callback for added shortcut as function.');
+    }
+
+    if (Array.isArray(variants) === false) {
+      throw new Error('Please define key variants for added shortcut as array of arrays with keys.');
+    }
+
+    const newShortcut = {
+      callback,
+      namespace,
+      runAction,
+      preventDefault,
+      stopPropagation,
+      relativeToNamespace,
+      position,
+    };
+
+    variants.forEach((variant) => {
+      const normalizedVariant = normalizeKeys(variant);
+      const hasVariant = SHORTCUTS.hasItem(normalizedVariant);
+
+      if (hasVariant) {
+        const shortcuts = SHORTCUTS.getItem(normalizedVariant);
+        let insertionIndex = shortcuts.findIndex(shortcut => shortcut.namespace === relativeToNamespace);
+
+        if (insertionIndex !== -1) {
+          if (position === 'before') {
+            insertionIndex -= 1;
+
+          } else {
+            insertionIndex += 1;
+          }
+
+        } else {
+          insertionIndex = shortcuts.length;
+        }
+
+        shortcuts.splice(insertionIndex, 0, newShortcut);
+
+      } else {
+        SHORTCUTS.addItem(normalizedVariant, [newShortcut]);
+      }
+    });
+  };
+
+  /**
+   * Add shortcuts to the context.
+   *
+   * @param {Array<object>} shortcuts List of shortcuts added to the context.
+   * @param {object} [options] Options for every shortcut.
+   * @param {Function} [options.callback] The callback.
+   * @param {object} [options.namespace] Namespace for shortcut.
+   * @param {object} [options.runAction]  Option determine whether assigned callback should be performed.
+   * @param {object} [options.stopPropagation=true] Option determine whether to stop event's propagation.
+   * @param {object} [options.preventDefault=true] Option determine whether to prevent default behavior.
+   * @param {object} [options.relativeToNamespace] Namespace name, relative which the shortcut is placed.
+   * @param {object} [options.position='after'] Position where shortcut is placed. It may be added before or after
+   * another namespace.
+   */
+  const addShortcuts = (shortcuts, options = {}) => {
+    shortcuts.forEach((shortcut) => {
+      objectEach(options, (value, key) => {
+        if (Object.prototype.hasOwnProperty.call(shortcut, key) === false) {
+          shortcut[key] = options[key];
+        }
+      });
+
+      addShortcut(shortcut);
+    });
+  };
+
+  /**
+   * Removes shortcut from the context.
+   *
+   * @param {Array<Array<string>>} variants A shortcut variant.
+   */
+  const removeShortcutByVariants = (variants) => {
```

Comment:
WDYT about naming change of the shortcuts? The project calls that keyboard shortcuts `variants`. It may be only me, but at first, glance is not obvious what the name means. Suggestions for consideration:
```js
removeShortcutByHotkeys = (hotkeys) => {}
// or 
removeShortcutByKeys = (keys) => {}
// or 
removeShortcutByKeyNames = (keyNames) => {}
```

Reply:
I've chosen `removeShortcutByKeys` name and changed it within https://github.com/handsontable/handsontable/pull/8955/commits/39e5f834d3b2f8787e8144c07a5fe9952764639a#diff-08f75a00996f54a5d780ffe1fb349e017b44c8cbe6fc6587a5abc5f5a03d99f6R166.

Reply:
@warpech Done within: 30829ab5f1f4848cd5462c661bf4c11ef87312df.